### PR TITLE
Fix type definition for dimension.group()

### PR DESCRIPTION
**Pull request by thatcort**
Hi, I believe this is a more correct type definition for the dimension.group() method.

### CODE REVIEW

File: index.d.ts [hunk 1]
```diff
@@ -65,9 +65,9 @@ declare namespace crossfilter {
     hasCurrentFilter(): boolean;
     top(k: number, offset?: number): TRecord[];
     bottom(k: number, offset?: number): TRecord[];
-    group<TGroupValue>(
-      groupValue?: (value: TValue) => NaturallyOrderedValue,
-    ): Group<TRecord, TValue, TGroupValue>;
+    group<TKey extends NaturallyOrderedValue, TGroupValue extends NaturallyOrderedValue>(
+      groupValue?: (value: TValue) => TKey,
+    ): Group<TRecord, Tkey, TGroupValue>;
```
Comment (esjewett) [inline index.d.ts]: I think there is a typo in the casing of `Tkey`

**Comment by esjewett** [comment]
So the idea here I that the *output* of the group accessor is the same type as the group key. The previous version assumed that the *input* of the group accessor (which is the same type as the output of the dimension accessor, FWIW) would be the type of the group key.

**Comment by thatcort** [comment]
Good catch on the casing of Tkey vs TKey. The PR should be updated now :)

**Comment by esjewett** [comment]
Target 1.4.6 - hope to get that out next week.

**Pull request merged**